tapfreighter/wallet.go

```diff
@@ -515,6 +515,12 @@ func (f *AssetWallet) FundPacket(ctx context.Context,
 
 	totalInputAmt := uint64(0)
 	for _, anchorAsset := range selectedCommitments {
+		// We only use the sum of all assets of the same ID to avoid
+		// counting passive assets as well.
+		if anchorAsset.Asset.ID() != fundDesc.ID {
+			continue
+		}
+
```

Comment:
I don't understand this change. Couldn't passive asset also have the same asset ID at this point?

Reply:
Not necessarily. If you look at `removeActiveCommitments`, we declare all assets as "passive" that don't have the same `TapCommitmentKey()`.
But I guess we should use the same comparison here as well, not just the ID. Will update.